### PR TITLE
Layer concept documentation

### PR DESCRIPTION
**Pull request by ikas**
## Summary

This PR refers to [GTC-277](https://gfw.atlassian.net/browse/GTC-277). It adds a new section to the concept docs regarding **Layers**.

## Testing instructions

If you feel like deep-diving into the contents of the PR, follow the steps below:

1. Clone the documentation repository and follow the [installation instructions on the README](https://github.com/resource-watch/doc-api#installation-and-usage).
2. Checkout the branch of this PR: `docs/layer-concept-docs`
3. Read the added concept docs for **layers**.

### CODE REVIEW

**Review by logoul — COMMENTED** [review]
@ikas , can we also mention here how /layer endpoint is different from the /contextual-layer endpoint and when users should be using one versus the other. Also, can we add a sentence on why it is important for each app to register their own layers instead of re-using layers defined by other apps. I think it also might be useful to add a clearer distinction between a dataset endpoint and a layer end point. One is used to define a dataset and is shared among the apps and the other is one more level of abstraction.

File: source/includes/concept/_layer.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Layer
+
+Apart from the possibility of fetching data from multiple data sources using a common interface, the RW API provides you with features to help you visualize such data. A **layer** is one of the resources that you can use to help you make custom visualizations for your data, and they play a very important role when visualizing datasets that contain geospatial data.
+
+A layer is a visual specification of how the data of a dataset can be rendered and styled on a map.
```
Comment (ethanroday) [inline source/includes/concept/_layer.md:5]: This is a good definition.

**Comment by tiagojsag** [comment]
Hey @logoul thanks for the feedback. I have a few questions and suggestions about your requests:

- about the difference between layers and contextual layers: I think it would be confusing for the reader if we explain the difference between layer and contextual layer, when currently there are no docs on contextual layers. So my suggestion here is to keep things simple, and don't mention contextual layers for now. In the future, when we tackle contextual layers, we can document it properly, and revisit this section to add a "if your use case is X, perhaps you are better off using contextual layers (add link to new shiny docs here)". Let me know if this makes sense

- add a sentence on why it is important for each app to register their own layers instead of re-using layers defined by other apps: I get the idea, but I think that perhaps the message should be tweaked to inform the user of the pros and cons of reusing existing vs creating new, and then let the user decide on their own. And perhaps this is best done on a broader level, as the same implications will be true for datasets, widgets, etc etc etc. So my suggestion here is adding another TODO [here](https://github.com/resource-watch/doc-api/blob/f5b8b0bbc668d2646d755891571bc1900704841c/source/includes/api/_before_you_get_started.md#L28-L27) and revisit it later. What do you think?

- add a clearer distinction between a dataset endpoint and a layer end point: this bit wasn't very clear to me, could you clarify, please? thanks!! :)

**Comment by logoul** [comment]
Hi @tiagojsag ,
1) I am generally OK with that, but a little bit afraid that we will forget to go back and update once we have the contextual layer documentation. 
2) I think this is very important to add here actually. People are very confused as to how and when to use layers endpoint. Do you think we can at least add very minimal set of pros/cons for registering app-specific layers. Also it would be great to add an example on how to register a layer for a vector tile cache and how to retrieve it.
3) Let me re-read the concept documents for layer and dataset endpoints. I want to make sure that users completely understand what they represent and how they should be used.

**Comment by tiagojsag** [comment]
Hi,

1- My suggestion here is to add a task on Jira to document the contextual layers endpoints, and add this note there, so us-in-a-few-months remember it.
2- These are just the concept docs, their purpose are to explain what a layer is. The "how to use" will come on the endpoints section, which I'm working on right now. Let's discuss this in more detail this afternoon.
3- Ok, let's also take some time this afternoon to answer any outstanding questions or comments you may have here.

Thanks

**Pull request merged**